pkg/plugins/resources/cargopackage/source.go

```diff
@@ -0,0 +1,33 @@
+package cargopackage
+
+import (
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+)
+
+// Source returns the latest npm package version
+func (cp CargoPackage) Source(workingDir string) (string, error) {
+	if len(cp.spec.IndexDir) == 0 && len(workingDir) > 0 {
+		cp.spec.IndexDir = workingDir
+	}
+
+	err := cp.Validate()
```

Comment:
This function could be moved to the function New() so we also validate the spec for condition and target